### ISSUE TITLE
Support tsconfig override

### ISSUE THREAD

**Issue by wemeetagain** [bbc/typescript-docs-verifier#22]
Hi again. I'm revamping the docs for https://github.com/libp2p/js-libp2p and noticing a subtle hiccup.
We want the typescript snippets in our docs to have different build options than our code.
A nice option would be to allow us to create a special `tsconfig.docs.json` with relaxed options and feed that to `typescript-docs-verifier` -- similarly to how a custom tsconfig can be passed to `tsc` and `ts-node` with `--project`.

As a specific example, we want docs snippets to support top-level await, but we don't want our source code to support top-level await.

```ts
import {createLibp2p} from 'libp2p'

const node = await createLibp2p({
  // docs explain various options here
})
```

Currently getting this error from `typescript-docs-verifier`:
```
✖ Error compiling example code block 17 in file doc/CONFIGURATION.md:
  ⨯ Unable to compile TypeScript:
      doc/CONFIGURATION.md → Code Block 17:3:14 - error TS1378: Top-level 'await' expressions are only allowed when the 'module' option is set to 'es2022', 'esnext', 'system', 'node16', or 'nodenext', and the 'target' option is set to 'es2017' or higher.
      
      3 const node = await createLibp2p({
                     ~~~~~
```

**Comment by paulbrimicombe** [comment]
Great idea @wemeetagain — I'll take a look.

**Comment by paulbrimicombe** [comment]
Hi @wemeetagain , I've just published v2.4.0 which adds a `--project` flag to the CLI which should fix this for you. Please let me know if it works / doesn't work!
